internal/services/shared/errors.go

```diff
@@ -116,13 +116,26 @@ func AsValidationError(err error) *ErrSchemaWriteDataValidation {
 
 type ConfigForErrors struct {
 	MaximumAPIDepth uint32
+	DebugTrace      *v1.DebugInformation
 }
 
+// DebugTraceErrorDetailsKey is the key used to store the debug trace in the error details.
+// The value is expected to be a string containing the proto text of a DebugInformation message.
+const DebugTraceErrorDetailsKey = "debug_trace_proto_text"
+
 func RewriteErrorWithoutConfig(ctx context.Context, err error) error {
-	return RewriteError(ctx, err, nil)
+	return rewriteError(ctx, err, nil)
 }
 
 func RewriteError(ctx context.Context, err error, config *ConfigForErrors) error {
+	rerr := rewriteError(ctx, err, config)
+	if config != nil && config.DebugTrace != nil {
+		spiceerrors.WithAdditionalDetails(rerr, DebugTraceErrorDetailsKey, config.DebugTrace.String())
```

Comment:
does `DebugTrace.String()` do the prototext conversion?

Reply:
Yes and this is tested